sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/arithmetic.scala

```diff
@@ -234,7 +234,17 @@ abstract class BinaryArithmetic extends BinaryOperator
   }
 
   override def dataType: DataType = (left.dataType, right.dataType) match {
-    case (DecimalType.Fixed(p1, s1), DecimalType.Fixed(p2, s2)) =>
+    case (DecimalType.Fixed(_p1, _s1), DecimalType.Fixed(_p2, _s2)) =>
+      // compatible with negative scale
+      val (p1, s1, p2, s2) = if (_s1 < 0 && _s2 < 0) {
```

Comment:
nit: are there any other methods that use ·`_xxx` for local vars name?

Reply:
good question, changed the name